virttest/qemu_devices/qdevices.py

```diff
@@ -89,7 +89,10 @@ def __init__(self, dev_type="QBaseDevice", params=None, aobject=None,
         self.params = OrderedDict()    # various device params (id, name, ...)
         if params:
             for key, value in six.iteritems(params):
-                self.set_param(key, value)
+                if key == "pcie_direct_plug":
+                    self.pcie_direct_plug = value == "yes"
```

Comment:
I understand how it comes, just not prefer to add such exception on fundamental definition.
How about `device.set_param("pcie_direct_plug", None)` so that it should be removed as I see in`def set_param()`:
```
elif value is None and option in self.params:
            del(self.params[option])
```

Reply:
@vivianQizhu, if we use `set_param` to save `pcie_direct_plug` property, when the value is not `None`, it will be passed into the qemu command line, because this property is not a qemu device option.

Reply:
@PaulYuuu Yes you are right, that might kill migration or repeatedly hotplug alike functions. But still I hope we can fix it but not introduce such weird exceptions. To make `pcie_direct_plug` a property is a good idea, I was thinking that to directly assign `device.pcie_direct_plug = False` when you get the param from cfg in qemu_vm.py, avoid calling `set_param`. But it seems that will require too much outside the pcie bus, which is also unexpected to me...Anyway I don't figure out a better solution so far, and I don't see any risk on your current solution. So I will approve it. Let's have @luckyh 's review, see if we can have a better idea.

Reply:
My understanding is a refactoring of bus topology management is something must to have, so this time let use the current solution as a workaround to fix the issue, thanks.